src/main/java/io/cdap/plugin/switchcase/route/Router.java

```diff
@@ -93,7 +93,10 @@ public void transform(StructuredRecord input, MultiOutputEmitter<StructuredRecor
       if (e.isDefaultValue()) {
         // the value does not match any rule in the port specification, emit it as a
         // defaulting value, based on the selected default handling option
-        emitDefaultValue(input, emitter);
+        if (Config.DefaultHandling.SKIP == emitDefaultValue(input, emitter)) {
+          // handle "skip" default port handling : do not emit to a port 
+           return;
+        }
       } else {
         emitNullValue(input, emitter);
       }
```

Comment:
I think adding a return after this line should be enough for the whole PR.

Reply:
yes, this works. I modified this in the next rev.